security/oidc-client-mutual-tls/src/main/resources/application.properties

```diff
@@ -11,6 +11,7 @@ quarkus.oidc.tls.key-store-password=${ks-pwd}
 quarkus.oidc.tls.trust-store-password=${ks-pwd}
 
 # TODO https://github.com/quarkusio/quarkus/issues/25972
+# We cannot test these properties before we configure all environment to run KeyCloak with BCFIPS provider in container
```

Comment:
You probably mean we cannot test these properties in FIPS? Because SunJCE can work without BCFIPS provider. It's provider itself after all. SunJCE is part of OpenJDK now, isn't it? I can see it f.e. in 21 https://github.com/openjdk/jdk21u/blob/master/src/java.base/share/classes/com/sun/crypto/provider/SunJCE.java

Question: isn't Keycloak container based on OpenJDK 17 based on UBI image https://issues.redhat.com/browse/RHELPLAN-110420, therefore PKCS12 with SunJCE provider should be supported without BCFIPS?

Reply:
I mean that we cannot test it because default Java providers work well with no error message and the same behaviour is for random provider name, `quarkus.oidc.tls.key-store-provider=Random1` , for example. We don't get an error message, because "these random values are ignored by the Java Keystore implementation for well-known key store types, it probably defaults to a default provider for dealing with the PKCS12 type if a given provider can't be found", as Sergey wrote [here](https://github.com/quarkusio/quarkus/issues/25972#issuecomment-1148746984). To test these properties properly we need to set unknown provider for Java, BCFIPS is an option because it's supported by KeyCloak.

So, for PKCS12 keystores we can set SunJCE provider. Keycloak 22 container is based on UBI 9 and OpenJDK 17.

Reply:
thanks for explanation